templates/docker-build.yml

```diff
@@ -0,0 +1,33 @@
+# Docker
+# Build a Docker image 
+# https://docs.microsoft.com/azure/devops/pipelines/languages/docker
+
+trigger:
+- master
+
+resources:
+- repo: self
+
+variables:
+  dockerfilePath: '{{ dockerfilePath }}'
+  tag: '$(Build.BuildId)'
+  
+  # Agent VM image name
+  vmImageName: 'ubuntu-latest'
+
+stages:
+- stage: Build
+  displayName: Build image stage
+  jobs:  
+  - job: Build
+    displayName: Build
+    pool:
+      vmImage: $(vmImageName)
+    steps:
+    - task: Docker@2
+      displayName: Build an image
+      inputs:
+        command: build
+        dockerfile: $(dockerfilePath)
+        tags: |
+          $(tag)
```

Comment:
I think the default tag itself is $(Build.BuildId); you don't need to add it explicitly maybe?

Reply:
Done. Removed

Reply:
No, let it be added explicitly. It'll be clearer for first time users reading the pipeline definition to understand stuff.